### ISSUE TITLE
minio-client: <ERROR> Unable to validate source s3 (while trying to move dir with that name)

### ISSUE THREAD

**Issue by Gandalf-the-Grey** [minio/mc#3209]
`minio-client mv -r s3 hive/images/`

( same for `./s3` )

## Expected Behavior
move recursively `s3` directory to destination

## Current Behavior
```
minio-client: <ERROR> Unable to validate source s3
```

## Steps to Reproduce (for bugs)

1. create a directory `s3`
2. put some files in there
3. attempt to move it to bucket
4. be surprised with a failure

## Context
I have a dir structure `[a-Z0-9][a-Z0-9]`

## Your Environment
Client and server run on same machine, Ubuntu 18.04 LTS
`minio-client version RELEASE.2020-04-25T00-43-23Z`
`minio version RELEASE.2020-04-28T23-56-56Z`

**Comment by kannappanr** [comment]
@Gandalf-the-Grey You cannot have a directory with the same name as an alias that already exists. Please use the workaround suggested above. Closing this issue as answered. Please feel free to reach out to us if you have further questions

**Comment by Gandalf-the-Grey** [comment]
If by alias you mean host name I got rid of it when I suspected it of being an issue:
`minio-client config host rm s3` so `minio-client config host list` no longer shows it
But that doesn't solve it. 

So when moving to minio millions of files I can move dirs `s0`, `s1`, `s2`, `s4`, `s5`, `s6`, `s7`, `s8` and `s9` but I can't move directory named `s3` because of reasons? ;-)

I'm not sure if you are aware how silly this is. Like if I couldn't have a directory named `xfs` on an xfs system.
